### ISSUE TITLE
Avro schema generation: allow override namespace with new `@AvroNamespace` annotation

### ISSUE THREAD

**Issue by fleetwoods** [FasterXML/jackson-dataformats-binary#310]
I was hoping to override the namespace used in the schemas generated by `AvroSchemaGenerator` for my POJOs, as opposed to the package name that is used (as it does not make sense to the consumers of my topic, outside of my application).

I cannot workout a means to do this, I tried `@JsonRootName(value = "ClassName", namespace = "my.domain")` but that is not picked up, is there any other means to achieve this?

Note this is not the same as `@JsonAlias`, I would like to control the namespace, not the `aliases`.

Code currently used:
```
    public static AvroSchema generateSchema(final Class<?> clazz) {
        final AvroSchemaGenerator gen = new AvroSchemaGenerator();
        gen.enableLogicalTypes();
        try {
            MAPPER.acceptJsonFormatVisitor(clazz, gen);
        } catch (JsonMappingException e) {
            throw new IllegalArgumentException("Cannot generate schema for: " + clazz, e);
        }
        return gen.getGeneratedSchema();
    }
```

**Comment by cowtowncoder** [comment]
As things are, `namespace` is unfortunately only used with XML backend.
But conceptually I can see how this could be useful for Avro as well.

If anyone has time to try a PR I'd be happy to help, but I do not have time to work on this directly myself.

**Comment by MichalFoksa** [comment]
Hi @cowtowncoder 

I think I found simple way how to override default Avro namespace value with a new, custom, annotation.

Please , see #324.

**Comment by cowtowncoder** [comment]
Will be released as part of 2.14.0.
